lib/slacks/connection.rb

```diff
@@ -330,7 +330,26 @@ def get_user_id_for_dm(dm)
 
 
     def api(command, params={})
-      response = http.post(command, params.merge(token: token))
+      params_with_token = params.merge(token: token)
+      page_limit = params.key?(:page_limit) ? params.delete(:page_limit) : MAX_PAGES
+      response = api_post command, params_with_token
+      fetched_pages = 1
+      while cursor = response.dig("response_metadata", "next_cursor") && fetched_pages < page_limit do
+        api_post(command, params_with_token.merge(cursor: cursor)).each do |key, value|
+          if value.is_a?(Array)
+            response[key].concat value
+          elsif value.is_a?(Hash)
+            response[key].merge! value
+          else
+            response[key] = value
+          end
+        end
+        fetched_pages += 1
+      end
```

Comment:
🤔  This method doesn't look like it returns a value. I believe the return value of `while` as an expression is always `nil` and `while` appears to be the last expression in this method

Reply:
😱 Good catch! I suppose we should return the response we've so carefully built up. 😅